### DISCUSSION TITLE
"Unavailable" and "Unknown" state for sensors

### DISCUSSION THREAD

**Discussion by giveyouup** [jason0x43/hacs-hubitat#200] [Q&A]
Is there a way to get "unavailable" or "Unknown" state to be used instead of the last known state? 

When one of my ZigBee sensors falls off the network due to a dead battery or other reason, the hubitat just reports the last known state in perpetuity it seems. The same status is pulled by this integration. Is there a way to make it so the status through this integration takes into account how long it's been since the last device update? And if it's been longer than X duration, the status on HA side is set to unknown or unavailable instead of whatever Hubitat thinks it is.

**Comment by jason0x43** [comment]
It’s possible to do that, although the integration would need to be careful not to flag something as “unavailable” that simply sends updates very infrequently.

Currently each device has a “last update time” entity that’s disabled by default. You can enable that, and then take the last update time into consideration on a device-specific basis.

**Reply by giveyouup** [reply]
Maybe this is more of a feature request? I am able to enable and view the last update time but I'm guessing being able to use the time stamp in flows and automations is gonna require some pretty involved coding. 

The end goal of this is to essentially mark devices that have fallen off the ZigBee or zwave network. I am using this sensor (https://github.com/jazzyisj/unavailable-entities-sensor) to display these and send me notifications when a certain device hasn't been heard from in, say a day. Open to other ideas. 

I do have device watchdog on the hubitat side that I could use for the same purpose but trying to use HA as front end for everything
